hs628_tools/Preprocess3_module.py

```diff
@@ -0,0 +1,63 @@
+import matplotlib
+import numpy as np
+import pandas as pd
+import matplotlib
+import matplotlib.pyplot as plt
+
+
+class vizsumstat:
+
+        def __init__ (self,csvPath,colname,col_catname):
+
+                self.df = pd.read_csv(csvPath)
+                self.colname = colname
+                self.col_catname = col_catname
+                #print(self.colname)
+                self.tuple = self.coltuple(self.df, self.colname, self.col_catname)
+                self.colgroupbyclass=self.group_by(self.tuple)
+
+                
+        def group_by(self,tpls,idx=1,merge=True):
+                d = dict()
+                for tpl in tpls:
+                        k = tpl[idx]
+                        v = d.get(k,tuple()) + (tpl[:idx]+tpl[idx+1:] if merge else (tpl[:idx]+tpl[idx+1:],))
+                        d.update({k:v})
+                return d
+
+        def coltuple(self,df,col,typpe):
+                #tuplle=[]
+                for i in range(len(df[col])):
+                        for k in range(len(df[typpe])):
+                                if i==k:
+                                        #newtuple=col[i],typpe[k]
+                                        #newtuple = ['col', 'typpe'].apply(tuple, axis=1)
+                                        newtuple = list(zip(df[col], df[typpe]))
+                                        #tuplle=tuplle+newtuple
+                                #print (newtuple)
+                        return newtuple
+                
+        def summarystat(self):
+                for i in range(len(self.colgroupbyclass)):
+                       avg=np.mean(self.colgroupbyclass[i])
+                       median=np.median(  self.colgroupbyclass[i])
+                       stdev=np.std(  self.colgroupbyclass[i])
+                       plt.hist(  self.colgroupbyclass[i], normed=True, bins=20)
+                       plt.title('The "Normal" Distribution with Mean & St. Devs.')
+                       plt.xlabel("Variables"); plt.ylabel("Frequency")
+                       plt.grid(True)
+                       plt.rc('grid', linestyle="dashed", color='grey')
+                       plt.show()
+                       #print(avg,median,stdev)
+                       print(i,"Mean = {0}".format(avg),"Median = {0}".format(median),"stdev = {0}".format(stdev))
+                return avg,median,stdev
+
+def main():
+        col=(input("input col name"))
```

Comment:
it's best to end your prompt with a colon and a space, as you do below.  Also, when you are just wanting plain string input as you are here, it is best to use ```raw_input``` rather than ```input```, or else explain in your prompt that the user will need to provide quotation marks around their input (or the program will crash unless they happen to give a name that matches one of your variables).

And the outer parentheses here are just clutter.